src/audio/copier/copier_gain.c

```diff
@@ -27,7 +30,7 @@ int copier_gain_set_params(struct comp_dev *dev, struct copier_gain_params *gain
 	/* Set basic gain parameters */
 	copier_gain_set_basic_params(dev, gain_params, ipc4_cfg);
 
-	switch (dd->dai->type) {
```

Comment:
how did this compiled before this patch?
I cannot see `dd` anywhere in the change,

Reply:
and if there were `dd` to begin with, then why we don't we take the pointers out from it locally?

Reply:
@ujfalusi `dd` was a function argument which is removed by this commit, so, yes, this commit seems to break compilation